src/cdk/v2/destinations/http/utils.js

```diff
@@ -165,14 +151,37 @@ const metadataHeaders = (contentType) => {
   }
 };
 
-const prepareBody = (payload, contentType) => {
+function removeUndefinedAndNullValuesDeep(obj) {
+  if (obj !== null && typeof obj === 'object' && !Array.isArray(obj)) {
+    return Object.entries(obj).reduce((acc, [key, value]) => {
+      const cleanedValue = removeUndefinedAndNullValuesDeep(value);
+      if (cleanedValue !== null && cleanedValue !== undefined) {
```

Comment:
we can use this
const { isDefinedAndNotNull } = require('@rudderstack/integrations-lib');

Reply:
why not move this to common utils in integration lib

Reply:
Replaced this function with an existing one in utils. Currently it is only being used in ga4_v2